### PR TITLE
RFC89: sql logging callback

### PR DESCRIPTION
**Pull request by elpaso**
RFC89: SQL query logging callback

### CODE REVIEW

File: doc/source/development/rfc/rfc89_sql_logging_callback.rst [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+in case of prepared queries the actual SQL after parameter
+substitutions should be sent.
+
+The callback will provide additional information about the executed
+query (if available):
+
+- error string message
+- number of affected/retrieved records
+- time taken to execute the query
+
```
Comment (rouault) [inline doc/source/development/rfc/rfc89_sql_logging_callback.rst:54]: It should be documented that the callback may be called by multiple threads in a concurrent way, and that the implementation of the callback should be robust to that (use of lock typically). That may arise in drivers that use multi-threading as an optimization implementation detail. And this is typically the case of the GeoPackage driver in its Arrow stream interface. Cf https://github.com/OSGeo/gdal/blob/master/ogr/ogrsf_frmts/gpkg/ogrgeopackagetablelayer.cpp#L7099.

Regarding the implementation in the GPKG driver, it should around https://github.com/OSGeo/gdal/blob/master/ogr/ogrsf_frmts/gpkg/ogrgeopackagetablelayer.cpp#L7121 install the callback set initially on the main dataset to the ``task->m_poDS`` internal dataset that will run in a dedicated thread.

**Comment by rouault** [comment]
you need to commit the updated index.rst to point to the RFC

File: doc/source/development/rfc/rfc89_sql_logging_callback.rst (hunk 1 above)
Comment (elpaso) [reply doc/source/development/rfc/rfc89_sql_logging_callback.rst:54]: I've shamelessly copy-pasted your sentence.

Comment (elpaso) [reply doc/source/development/rfc/rfc89_sql_logging_callback.rst:54]: About the implementation, I've installed the callback this way: https://github.com/elpaso/gdal/commit/4b1bf394e5af24e7758aa31c7162ee61f9a768dd#diff-125788fb2cb0d32e396c0af12cb5bbc19b12a006603dddc29236a3afdb9c8096R4342

Comment (rouault) [reply doc/source/development/rfc/rfc89_sql_logging_callback.rst:54]: > I've installed the callback this way

My point regarding the multi-threaded optimization was that, to instrument everything, the callback also needs to be propagated by the driver itself from the main dataset to the auxiliary datasets it opens on the same file, and that should be done around https://github.com/OSGeo/gdal/blob/master/ogr/ogrsf_frmts/gpkg/ogrgeopackagetablelayer.cpp#L7121

Comment (elpaso) [reply doc/source/development/rfc/rfc89_sql_logging_callback.rst:54]: See https://github.com/elpaso/gdal/commit/4e3f0e8eb742599320ba9e1da4eb4ae988004a77

**Comment by rouault** [comment]
@elpaso You can update the "Voting history" section with "+1 from PSC members MateuszL, JukkaR, TamasS and EvenR", and change its status to "Adopted, implemented" (well, "implemented" only after the implementation PR will have been merged)

**Comment by elpaso** [comment]
@rouault done, I'll make the iplementation PR after rebase and reformat, probably tomorrow.

File: doc/source/development/rfc/rfc89_sql_logging_callback.rst [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+=============================================================
+RFC 89: SQL logging callback
+=============================================================
+
+============== =============================================
+Author:        Alessandro Pasotti
+Contact:       elpaso @ itopen.it
+Started:       2022-Nov-30
+Status:        Draft
```
Comment (rouault) [inline doc/source/development/rfc/rfc89_sql_logging_callback.rst:11]:
```suggestion
Status:        Adopted, implemented
```

**Pull request merged**